### PR TITLE
design(1314): add `--disable-image-pull` option

### PR DESCRIPTION
**Pull request by MysticDoll**
## Context

We are now developing `sd-local` and it always pulls build image each execution.
Therefore this PR suggests adding the option to disable that behaviour to make users run jobs on docker images in local machine.

## Objective
- add `--always-pull-disable` option.

## References
- #1314 
- screwdriver-cd/sd-local/pull/11

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by s-yoshika** [comment]
When we can use this option? The build with this option can't ensure the same build as the actual SD build, because SD builds always pull images.

**Comment by yuichi10** [comment]
I think sometime user want to use this option for testing image of build.

**Comment by s-yoshika** [comment]
I see, but it's hard to guess the usage from this name.

**Comment by MysticDoll** [comment]
> it's hard to guess the usage from this name.

how about `--use-local-image`?

**Comment by jithine** [comment]
can we make the default to use cached image ? like how `docker run` works.

**Comment by s-yoshika** [comment]
> can we make the default to use cached image ? like how docker run works.

I concern it can't guarantee the reproducibile build, for example the image which has latest tag might not be latest without pulling. And the build environment can be different from those of actual Screwdriver build.

File: design/sd-local.md [hunk 1]
```diff
@@ -135,6 +135,7 @@ $ sdlocal build [job-name] [options]
 - `--artifacts-dir [path]` Path to the host side directory which is mounted into `$SD_ARTIFACTS_DIR`. (default: `./sd-artifacts`)
 - `-m, --memory [size]` Set memory size which Build Container can use. Either b, k, m, g can be used as a size unit. (default: ?)
 - `--src-url [repository url]` Set repository URL which is to build when user use the remote repository without local files.
+- `--always-pull-disable` Disable `sd-local` from always pulling build image.
```
Comment (jithine) [inline design/sd-local.md]: can we call it `--disable-image-pull` ?

Comment (MysticDoll) [reply design/sd-local.md]: I fixed it. It seems better than before. thx

**Pull request merged**